extensions/wikia/Recirculation/RecirculationController.class.php

```diff
@@ -15,41 +17,116 @@ public function init() {
 	}
 
 	public function discussions() {
+		global $wgLanguageCode;
+
+		$showZeroState = false;
 		$cityId = $this->request->getVal( 'cityId', null );
+		$limit = $this->request->getVal( 'limit', 5 );
+		$ignoreWgEnableRecirculationDiscussions =
+			$this->request->getVal( 'ignoreWgEnableRecirculationDiscussions', false );
+		$sortKey =
+			$this->request->getVal( 'latest', false )
+				? DiscussionsDataService::DISCUSSIONS_API_SORT_KEY_LATEST
+				: DiscussionsDataService::DISCUSSIONS_API_SORT_KEY_TRENDING;
 
 		if ( !empty( $cityId ) && !is_numeric( $cityId ) ) {
 			throw new InvalidParameterApiException( 'cityId' );
 		}
 
-		if ( RecirculationHooks::canShowDiscussions( $cityId ) ) {
-			$discussionsDataService = new DiscussionsDataService( $cityId );
-			$posts = $discussionsDataService->getData( 'posts' )['posts'];
+		if ( RecirculationHooks::canShowDiscussions( $cityId, $ignoreWgEnableRecirculationDiscussions ) ) {
+			$discussionsDataService = new DiscussionsDataService( $cityId, $limit );
+			$posts = $discussionsDataService->getData( 'posts', $sortKey )['posts'];
+
+			$discussionsUrl = "$discussionsDataService->server/d/f";
 
-			if ( count( $posts ) > 0 ) {
-				$discussionsUrl = "$discussionsDataService->server/d/f";
+			$postObjects = [];
 
-				$postObjects = [];
+			if (empty($posts)) {
+				$showZeroState = true;
+			}
 
-				foreach ( $posts as $post ) {
-					$postObjects[] = $post->jsonSerialize();
-				}
+			foreach ( $posts as $post ) {
+				$postObjects[] = $post->jsonSerialize();
+			}
 
-				$this->response->setCacheValidity( WikiaResponse::CACHE_VERY_SHORT );
-				$this->response->setData( [
-					'title' => wfMessage( 'recirculation-discussion-title' )->plain(),
-					'linkText' => wfMessage( 'recirculation-discussion-link-text' )->plain(),
-					'discussionsUrl' => $discussionsUrl,
-					'posts' => $postObjects,
-				] );
-				return true;
+			if ($wgLanguageCode === 'en') {
+				//This is temporary to render new discusions card on en wikis (templates in php and mustache have the same name)
+				$this->response->setTemplateEngine( WikiaResponse::TEMPLATE_ENGINE_PHP );
 			}
+			$this->response->setCacheValidity( WikiaResponse::CACHE_VERY_SHORT );
+			$this->response->setData( [
+				'title' => wfMessage( 'recirculation-discussion-title' )->plain(),
+				'linkText' => wfMessage( 'recirculation-discussion-link-text' )->plain(),
+				'discussionsUrl' => $discussionsUrl,
+				'posts' => $postObjects,
+				'showZeroState' => $showZeroState,
+			] );
+
+			return true;
 		}
 
 		return false;
 	}
 
+	public function footer() {
+		global $wgSitename, $wgCityId, $wgLanguageCode;
+
+		// Language code check is temporary to supress MCF for old discussions on non-en wikis;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)